### PR TITLE
Podman build

### PR DESCRIPTION
**Pull request by dbnicholson**
This is on top of learningequality/kolibri-installer-android#108. I was going to send this upstream, but maybe we should test it out here first. I successfully did a few podman and docker builds with it.

### CODE REVIEW

**Comment by danigm** [comment]
I've been testing to build using podman with this change and it's working correctly for me.

**Comment by dbnicholson** [comment]
Rebased this now. We'll see if it breaks the CI...

File: scripts/rundocker.sh [hunk 1]
```diff
@@ -37,6 +30,36 @@ RUN_OPTS=(
...
+fi
+
+# If the container user is root, mount the cache at /root. Otherwise,
+# set HOME since there's likely no account with that UID in the image.
+# The user's home directory is where all the intermediate build outputs
+# (e.g., ~/.local/share/python-for-android and ~/.gradle) are stored.
+if [ "$BUILD_UID" -eq 0 ]; then
+  BUILD_CACHE_PATH=/root
+else
+  BUILD_CACHE_PATH=/cache
```
Comment (manuq) [inline scripts/rundocker.sh:51]: Testing here. The changes in the `--user` parameter works for me. But the change in `BUILD_CACHE_PATH` is making the caching worse. I ran the build a second time and is taking ages. I will keep testing.

Comment (manuq) [reply scripts/rundocker.sh:51]: > Testing here. The changes in the `--user` parameter works for me. But the change in `BUILD_CACHE_PATH` is making the caching worse. I ran the build a second time and is taking ages. I will keep testing.

I was wrong! This branch in fact reduces the successive builds to 1 minute. Great!!

**Pull request merged**